share/config.fish

```diff
@@ -256,6 +202,37 @@ end
 #
 
 if status --is-login
+    # OS X-ism: Call path_helper and set its output to $PATH and $MANPATH.
+    if command -sq /usr/libexec/path_helper
+        set -l lines (/usr/libexec/path_helper -c)
+        for line in $lines
+            # Match lines of the form:
+            #
+            #   setenv KEY value
+            #
+            # ignoring whitespace before and after value, and an
+            # optional trailing semicolon.
+            if set -l match (string match -r '^\s*setenv\s+([A-Z]+)\s+(.*?)\s*;?\s*$' -- $line)
+                set -l key $match[2]
+                # Only set PATH and MANPATH, the two environment
+                # variables that path_helper is documented to emit.
+                if contains -- $key PATH MANPATH
+                    set -l value $match[3]
+                    # Strip any surrounding quotes from $value. Note
+                    # that we don't actually want to worry about
+                    # unescaping escaped quotes in $value, as
+                    # path_helper (buggily) doesn't bother escaping
+                    # quotes.
+                    if string match -qr '^([\'"]).*\1$' -- $value
+                        set value (string match -r '^.(.*).$' $value)[2]
```

Comment:
You can use `string trim` with `-c` to specify characters to strip from start/end instead of a regex, fyi.

Reply:
That doesn't quite do what is needed, though -- it would (using [] to denote strings) turn ["'foo bar""] into [foo bar], where it should just be ['foo bar"].